### DISCUSSION TITLE
Limit Active Power

### DISCUSSION THREAD

**Discussion by renatod47** [wlcrs/huawei_solar#402] [Ideas]
Hello,
Im interested in limiting active power, but there's no option for it.
When I try to limit the grid power feed, nothing happens. I saw here:
https://gitlab.com/Emilv2/huawei-solar/-/blob/master/src/huawei_solar/registers.py
That It's possible to edit the modbus variables that I am trying to reach, and as I have the modbus address list, I wanted to try, but I couldn't reach the file itself.
Any ideas?

**Comment by dim123dim123** [comment]
i have same problem 

as we see there is ```python
rn.ACTIVE_POWER_CONTROL_MODE: U16Register(rv.ActivePowerControlMode, 1, 47415, 1, writeable=True),
    rn.MAXIMUM_FEED_GRID_POWER_WATT: I32Register("W", 1, 47416, 2, writeable=True),
    rn.MAXIMUM_FEED_GRID_POWER_PERCENT: I16Register("%", 10, 47418, 1, writeable=True),
But no possible set ZERO Power mode. 

Is there some requirements ? Power Meter is needed for Grid-tie power limit ?

**Reply by Roving-Ronin** [reply]
@renatod47 
Sorry are you trying to limit the EXPORT of power back to the grid? 
Assuming your trying to do this, as you have a electricity provider that charges you to export power at certain times of day, or if there is too much power in the overall grid? If so, there's the following services that WLCRS has provided:

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/62a29985-4f61-4736-9f70-4e812929af1d)

Using these you can create an automation that says, for example if you need to limit export at 10am, then try:
![image](https://github.com/wlcrs/huawei_solar/assets/108674933/cd2bf10d-192b-4c65-9786-ba80567890fe)

Later on when you need to return exports to unlimited, instead call:
![image](https://github.com/wlcrs/huawei_solar/assets/108674933/bdd89960-942f-4645-9cc7-c4eed3d0f7ba)

If you have an electricity retailer like Amber Energy (Australia) who provide a HA integration that allows access to live prices as to your per kWh cost to import and export, you could select that as the 'Trigger' in the automation and add conditions like 'if export cost is negative$ / less than $0.05' then call the Action to restrict power being exported to the grid. In this way you would have the ability to actively determine when its profitable or not to export to the grid and then stop/allow exports.

Note: 
1) If you have multiple inverters in a cascade, you select the Primary/Master inverter in the cascade, it in turn controls the actions of the other inverters in the cascade chain.
2) Being able to use these services requires that you have a consumption meter installed and wired to your inverter (i.e DDSU-666 for single phase or a DSTU-666H for 3 phase).  **<---**  @dim123dim123

**Reply by ripcam21** [reply]
Hi,

I have been successfully limited PV output to grid using the service - 'Limit the power fed to the grid to percentage'.  As I use Amber, this prevents me from being charged for exporting PV excess.  However, sometime grid energy is also negative or extremely cheap and I would like the home to only use grid; not PVs or battery during this.  What service would do this?  How would the automation in you view work?  I would like to use a condition which states if price below 5c, charge battery and power home.  thoughts?

Any help would be appreciated.

Cheers,
Cameron.
